dashboard/src/components/AppUpgrade/AppUpgrade.tsx

```diff
@@ -48,17 +50,28 @@ function AppUpgrade() {
   const repoName = repo?.metadata?.name || app?.availablePackageRef?.context?.namespace;
   const repoNamespace = repo?.metadata?.namespace || app?.availablePackageRef?.context?.namespace;
 
+  const [pluginObj] = useState(
+    selected.availablePackageDetail?.availablePackageRef?.plugin ??
+      ({ name: plugin?.split("-")[0], version: plugin?.split("-")[1] } as Plugin),
+  );
+
   useEffect(() => {
-    dispatch(actions.apps.getApp(cluster, namespace, releaseName));
-  }, [dispatch, cluster, namespace, releaseName]);
+    dispatch(actions.apps.getApp(cluster, namespace, releaseName, pluginObj));
+  }, [dispatch, cluster, namespace, releaseName, pluginObj]);
 
   useEffect(() => {
-    if (repoNamespace && app?.availablePackageRef?.identifier && app?.currentVersion?.pkgVersion) {
+    if (
+      repoNamespace &&
+      app?.availablePackageRef?.identifier &&
+      app?.currentVersion?.pkgVersion &&
+      app?.availablePackageRef?.plugin
```

Comment:
Possible `.isValid()` again (not something needed, just wondering what you think, rather than checking multiple fields in multiple places).

Reply:
As in other comment
> wrt the first comment, yep, fixed. Once we use the ...packageRef we no longer need this check